animations/lib/src/basics/06_tween_sequence.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+
+class TweenSequenceDemo extends StatefulWidget {
+  static const String routeName = '/basics/chaining_tweens';
+
+  @override
+  _TweenSequenceDemoState createState() => _TweenSequenceDemoState();
+}
+
+class _TweenSequenceDemoState extends State<TweenSequenceDemo>
+    with SingleTickerProviderStateMixin {
+  static const Duration duration = Duration(seconds: 1);
+  AnimationController controller;
+  Animation<Color> animation;
+
+  static final List<Color> colors = [
+    Colors.red,
+    Colors.orange,
+    Colors.yellow,
+    Colors.green,
+    Colors.blue,
+    Colors.indigo,
+    Colors.purple,
+  ];
+
+  void initState() {
+    super.initState();
+
+    final sequenceItems = <TweenSequenceItem<Color>>[];
+
+    for (var i = 0; i < colors.length; i++) {
+      final beginColor = colors[i];
+      final endColor = colors[(i + 1) % colors.length];
+      final weight = 1 / colors.length;
+
+      sequenceItems.add(
+        TweenSequenceItem<Color>(
+          tween: ColorTween(begin: beginColor, end: endColor),
+          weight: weight,
+        ),
+      );
+    }
+
+    controller = AnimationController(duration: duration, vsync: this);
+    animation = TweenSequence<Color>(sequenceItems).animate(controller)
+      ..addListener(() => setState(() {}));
```

Comment:
A number of the basic examples currently in the app (custom tween and animation controller, for example) use setState as the mechanism for putting the value of the animation to work, even though they could be using AnimatedBuilder.

Was using AnimatedBuilder as the "default" way to rebuild with an animation considered? Is once of the two approaches noticeably better than the other?

Reply:
My intention was to introduce AnimatedBuilder later, but now that you mention it, it makes sense to introduce it sooner. 

I'll switch up the order to make AnimatedBuilder show up earlier.

Reply:
SGTM.

FWIW, though, don't ever be afraid to disagree with me on questions like this. Whenever you see me write out a multi-line question in a thinking-out-loud style like this, I'm genuinely interested in other folks' opinions.

You've already saved me from my own bad ideas several times. :)